src/System.Net.Http/src/System/Net/Http/HttpContent.cs

```diff
@@ -568,5 +580,17 @@ private void CheckSize(int countToAdd)
                 }
             }
         }
+
+        private struct EncodingPreamblePair
```

Comment:
What'd you have in mind? Two arrays (one for the encodings and one for the `byte[]` preambles, that share the same indexes)? Or just one array of `byte[]` preambles with known indexes to get the associated encoding?